lib/buffer.js

```diff
@@ -335,6 +336,14 @@ Buffer.byteLength = byteLength;
 
 // toString(encoding, start=0, end=buffer.length)
 Buffer.prototype.toString = function(encoding, start, end) {
+  // .toString() can be called automatically on non-Buffers.
```

Comment:
To get here you'd have to call `toString.{call,apply}()` or to copy it to the prototype of another constructor, etc. Either way you're hosed. Like @domenic mentions, just throw a `TypeError` if the `instanceof` check fails.